server/graphql/v2/input/AccountReferenceInput.js

```diff
@@ -93,3 +94,36 @@ export const fetchAccountWithReference = async (
   }
   return collective;
 };
+
+/**
+ * Retrieves accounts for given ids or slugs
+ *
+ * @param {object} inputs - object containing slugs or ids of the collectives
+ * @param {object} params
+ *    - throwIfMissing: throws an exception if a collective is missing for a given id or slug
+ */
+export const fetchAccountsWithReferences = async (inputs, { throwIfMissing = false }) => {
+  const getSQLConditionFromAccountReferenceInput = inputs => {
+    const conditions = [];
+    inputs.forEach(input => {
+      if (input.id && typeof input.id === 'string') {
+        const id = idDecode(input.id, 'account');
+        conditions.push({ id });
+      } else if (input.legacyId || typeof input.id === 'number') {
+        conditions.push({ id: input.legacyId || input.id });
```

Comment:
I don't see how we could end up with different types here. The expected input type is an `AccoutReferenceInput`, if we're passing something else then we're misusing the function.

```suggestion
      if (input.id) {
        conditions.push({ id: idDecode(input.id, 'account') });
      } else if (input.legacyId) {
        conditions.push({ id: input.legacyId });
```